packages/components/src/components/search/search.tsx

```diff
@@ -319,6 +337,46 @@ export class Search implements ComponentInterface {
         });
     }
 
+    /**
+     * Get Mapbox Places results.
+     *
+     * @param {string} query
+     * @return {Promise<any>}
+     */
+    private getMapboxSearchResults(query: string): Promise<any> {
+        if (this.mapbox && this.mapboxAccessToken) {
+            if (query) {
+                return new Promise((resolve) => {
+                    const url = `https://api.mapbox.com/search/searchbox/v1/suggest?q=${query}&session_token=[GENERATED-UUID]&access_token=${this.mapboxAccessToken}`;
+
+                    fetch(url)
+                        .then((response) => {
+                            return response.json();
+                        })
+                        .then((result) => {
+                            const places = result.suggestions.map((result) => ({
+                                id: result.mapbox_id,
+                                type: 'Feature',
+                                properties: {
+                                    type: 'mapbox_places',
+                                    placeId: result.mapbox_id,
+                                    name: result.name,
+                                    subtitle: result.place_formatted || '',
+                                    floor: 0
+                                }
+                            }));
+                            resolve(places);
+                        })
+                        .catch(err => {
+                            console.log('Error: ', err);
```

Comment:
Should we handle this better? Maybe just resolve with an empty array.

Reply:
I changed it now to `resolve([])`. Is this enough, or should we resolve the Promise?

Reply:
Do you think it should be refactored, or should we keep the resolve for now, and maybe some cases will be revealed after the testing of this feature?

Reply:
I suggest that we keep the resolve with the empty array for now.